### PR TITLE
Clarify resource annotations behavior

### PR DESCRIPTION
**Pull request by kfaseela**
For more details see: https://github.com/istio/istio/pull/37825#issuecomment-1068499203

Signed-off-by: Faseela K <faseela.k@est.tech>

- [ X] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

**Review by howardjohn — APPROVED** [review]

File: content/en/docs/setup/additional-setup/sidecar-injection/index.md [hunk 1]
```diff
@@ -219,7 +219,35 @@ In general, any field in a pod can be set. However, care must be taken for certa
...
+For example, see the below incomplete resources annotation configuration and the corresponding injected resource settings:
+
+{{< text yaml >}}
+spec:
+  template:
+    metadata:
+      annotations:
+        sidecar.istio.io/proxyCPU: "200m"
+        sidecar.istio.io/proxyMemoryLimit: "5Gi"
+{{< text yaml >}}
```
Comment (ericvn) [inline content/en/docs/setup/additional-setup/sidecar-injection/index.md:236]: I suspect you want this be be a /text. If you look at the netlify output, this section as a lot of weird characters.

File: content/en/docs/setup/additional-setup/sidecar-injection/index.md [hunk 2]
```diff
@@ -219,7 +219,35 @@ In general, any field in a pod can be set. However, care must be taken for certa
...
+  - name: istio-proxy
+    resources:
+      limits:
+        memory: 5Gi
+      requests:
+        cpu: 200m
+        memory: 5Gi
+      securityContext:
+        allowPrivilegeEscalation: false
+{{< text yaml >}}
```
Comment (ericvn) [inline content/en/docs/setup/additional-setup/sidecar-injection/index.md:250]: Same as comment above. Make a /text.

File: content/en/docs/setup/additional-setup/sidecar-injection/index.md [hunk 3]
```diff
@@ -219,7 +219,35 @@ In general, any field in a pod can be set. However, care must be taken for certa
   it is recommended to set the `image` to `auto` which will cause the sidecar injector to automatically select the image to use.
 * Some fields in `Pod` are dependent on related settings. For example, CPU request must be less than CPU limit. If both fields are not configured together, the pod may fail to start.
 
-Additionally, certain fields are configurable by [annotations](/docs/reference/config/annotations/) on the pod, although it is recommended to use the above approach to customizing settings.
+Additionally, certain fields are configurable by [annotations](/docs/reference/config/annotations/) on the pod, although it is recommended to use the above approach to customizing settings. Additional care must be taken for certain annotations:
+
+* If `sidecar.istio.io/proxyCPU` is set, make sure to explicitly set `sidecar.istio.io/proxyCPULimit` too. Otherwise the missing resource will be set as unlimited.
```
Comment (ericvn) [inline content/en/docs/setup/additional-setup/sidecar-injection/index.md:224]: Remove the `too` in this and the next sentence.

File: content/en/docs/setup/additional-setup/sidecar-injection/index.md (hunk 2 above)
Comment (kfaseela) [reply content/en/docs/setup/additional-setup/sidecar-injection/index.md]: Done

File: content/en/docs/setup/additional-setup/sidecar-injection/index.md (hunk 3 above)
Comment (kfaseela) [reply content/en/docs/setup/additional-setup/sidecar-injection/index.md]: Removed

File: content/en/docs/setup/additional-setup/sidecar-injection/index.md [hunk 4]
```diff
@@ -219,7 +219,35 @@ In general, any field in a pod can be set. However, care must be taken for certa
...
+For example, see the below incomplete resources annotation configuration and the corresponding injected resource settings:
+
+{{< text yaml >}}
+spec:
+  template:
+    metadata:
+      annotations:
+        sidecar.istio.io/proxyCPU: "200m"
+        sidecar.istio.io/proxyMemoryLimit: "5Gi"
+{{< /text>}}
```
[the remark this answers is not in the document]
Comment (kfaseela) [reply content/en/docs/setup/additional-setup/sidecar-injection/index.md]: Done

**Comment by ericvn** [comment]
One final comment, you have `Otherwise the missing resource will be set as unlimited.` I don't think the missing resource is being set, I think you want to say something like `Otherwise the resource's memory/cpu (depending on the line) limit will be set as unlimited.`

**Review by ericvn — APPROVED** [review]

**Pull request merged**